### ISSUE TITLE
Public RTSP sources as example

### ISSUE THREAD

**Issue by grigory-lobkov** [grigory-lobkov/rtsp-camera-view#3]
Since 10.05.2018 added some public RTSP sources as examples
- Hessdalen
http://freja.hiof.no:1935/rtplive/_definst_/hessdalen03.stream/playlist.m3u8
- USA golf
http://wmccpinetop.axiscam.net/mjpg/video.mjpg
- Highway
rtsp://170.93.143.139/rtplive/470011e600ef003a004ee33696235daa
- Semerkand TV
rtmp://semerkandglb.mediatriple.net:1935/semerkandliveedge/semerkand2

I'd like the source to have good and bad resolution sources to show program capabilities.

**Do you have any offers?**

**Comment by hblanken** [comment]
rtsp://freja.hiof.no:1935/rtplive/_definst_/hessdalen03.stream

**Comment by abhijeet2431** [comment]
https://g33ktricks.blogspot.com/p/the-rtsp-real-time-streaming-protocol.html
It has huge list of free RTSP links for testing.

**Comment by grigory-lobkov** [comment]
I didn't find any working streams(for vlc 2.1.5) from links above, sorry. 
I have found a good [source](https://www.insecam.org/en/byrating/) of public cameras.

Since 25.04.2020 the example cameras are
- Tokyo
http://210.148.114.53/-wvhttp-01-/GetOneShot?image_size=640x480&frame_count=1000000000

- Krasnodar, Sochi
http://158.58.130.148:80/mjpg/video.mjpg

- Antwerpen, Antwerpen
http://81.83.10.9:8001/mjpg/video.mjpg

- Colorado, Glenwood Springs
http://208.72.70.171:80/mjpg/video.mjpg

**Comment by tabishimran** [comment]
security cams
http://46.16.226.181:88/?action=stream
http://79.132.232.174:8080/?action=stream
http://46.151.101.158:8081/?action=stream
http://90.66.111.157:8080/?action=stream
http://91.49.234.155:9000/?action=stream
http://46.151.101.149:8081/?action=stream
http://78.99.31.149:8081/?action=stream
http://46.151.101.134:8082/?action=stream
equipment cams
http://68.3.132.170:8080/?action=stream
http://129.10.131.166:8080/?action=stream
http://89.39.27.182:8080/?action=stream
http://66.130.223.168:8080/?action=stream
http://98.30.39.160:8080/?action=stream

**Comment by BraINstinct0** [comment]
> security cams
> http://46.16.226.181:88/?action=stream
> http://79.132.232.174:8080/?action=stream
> http://46.151.101.158:8081/?action=stream
> http://90.66.111.157:8080/?action=stream
> http://91.49.234.155:9000/?action=stream
> http://46.151.101.149:8081/?action=stream
> http://78.99.31.149:8081/?action=stream
> http://46.151.101.134:8082/?action=stream
> equipment cams
> http://68.3.132.170:8080/?action=stream
> http://129.10.131.166:8080/?action=stream
> http://89.39.27.182:8080/?action=stream
> http://66.130.223.168:8080/?action=stream
> http://98.30.39.160:8080/?action=stream

where did you get those?

**Comment by erikmartinjordan** [comment]
I couldn't find any working RTSP streams from the links above. Someone knows a reliable RTSP public link?

**Comment by crossworth** [comment]
@erikmartinjordan 
rtsp://wowzaec2demo.streamlock.net/vod/mp4:BigBuckBunny_115k.mov

**Comment by bobby20180331** [comment]
> http://68.3.132.170:8080/?action=stream
> http://129.10.131.166:8080/?action=stream
> http://89.39.27.182:8080/?action=stream
> http://66.130.223.168:8080/?action=stream
> http://98.30.39.160:8080/?action=stream

equipment cameras work！

**Comment by tabishimran** [comment]
@bobby20180331 , yea most of these links stop working after a while, I wrote a crawler to index these links a while ago because I needed some links to test something I was building.

**Comment by botuhok** [comment]
Kirchhoff Institute for Physics, Germany
http://pendelcam.kip.uni-heidelberg.de/mjpg/video.mjpg

Blanton Bottling, Kentucky, USA 
http://camera.buffalotrace.com/mjpg/video.mjpg

Butovo, Moscow, Russia
http://camera.butovo.com/mjpg/video.mjpg

Purdue Engineering Mall, USA
http://webcam01.ecn.purdue.edu/mjpg/video.mjpg

Tokyo, Japan
http://61.211.241.239/nphMotionJpeg?Resolution=320x240&Quality=Standard

Japan
http://vetter.viewnetcam.com:50000/nphMotionJpeg?Resolution=640x480&Quality=Clarity

Tampere Hacklab, Finland
http://tamperehacklab.tunk.org:38001/nphMotionJpeg?Resolution=640x480&Quality=Clarity

Takemoto Piano, Osaka, Japan
http://takemotopiano.aa1.netvolante.jp:8190/nphMotionJpeg?Resolution=640x480&Quality=Standard&Framerate=30

Richtung West, Switzerland
http://clausenrc5.viewnetcam.com:50003/nphMotionJpeg?Resolution=320x240

Soltorget Pajala, Sweden
http://195.196.36.242/mjpg/video.mjpg

Tsumago, Japan
http://honjin1.miemasu.net/nphMotionJpeg?Resolution=640x480&Quality=Standard

Mohouli Park, Hilo, Hawaii
http://67.53.46.161:65123/mjpg/video.mjpg

Rhein-Taunus-Krematorium, Germany
http://webcam.rhein-taunus-krematorium.de/mjpg/video.mjpg

Kaiskuru Skistadion, Norway  
http://77.222.181.11:8080/mjpg/video.mjpg

Community Hospital San Bernardino, USA   
http://webcam.mchcares.com/mjpg/video.mjpg?timestamp=1566232173730

Warrenton, OR, USA   
http://47.51.131.147/-wvhttp-01-/GetOneShot?image_size=1280x720&frame_count=1000000000

**Comment by botuhok** [comment]
you can find it in google with:

_inurl:/mjpg/video.mjpg_

**Comment by BraINstinct0** [comment]
Think it'll be possible to set up a site that shows the currently available sites?

**Comment by tomaszek92** [comment]
rtsp://rtsp.stream/movie
rtsp://rtsp.stream/pattern

**Comment by rtspstream** [comment]
Thanks for posting these URLs! I am the creator of https://www.octostream.com and I put together this service after struggling to find a reliable RTSP Stream to test my software application. I would like to hear from the community on how you're using these URLs and if there's any feature that you'd like to see. Considering the server/bandwidth costs, I also plan to introduce a paid plan (while still maintaining the current features as part of the free plan). What are some of the features you'd be happy to pay for? Please reach out to me on dave@octostream.com with your feedback and suggestions, thank you!

**Comment by fpaupier** [comment]
> Thanks for posting these URLs! I am the creator of https://rtsp.stream and I put together this service after struggling to find a reliable RTSP Stream to test my software application. I would like to hear from the community on how you're using these URLs and if there's any feature that you'd like to see. Considering the server/bandwidth costs, I also plan to introduce a paid plan (while still maintaining the current features as part of the free plan). What are some of the features you'd be happy to pay for? Please reach out to me on [dave@rtsp.stream](mailto:dave@rtsp.stream) with your feedback and suggestions, thank you!

Thanks a lot for this service Dave! I used your `rtsp` to test a videostream project (performing rtsp towards mp4 conversion) (https://github.com/horgh/videostreamer/pull/23)

**Comment by hdanielmora** [comment]
> dave

hi Dave,  Thx a lot. I just started to play around with it. perhaps an authenticated version could be nice.

**Comment by necrodome** [comment]
I am building a machine learning pipeline that involves RTSP streams and needed sample streams to test on. It's pretty easy to run your own public service with the help of [https://github.com/aler9/rtsp-simple-server](rtsp-simple-server). Here are some sample streams to test if you need.

- [rtsp://rtspstream.com/pattern](rtsp://rtspstream.com/pattern) 
- [rtsp://rtspstream.com/ball](rtsp://rtspstream.com/ball)
- [rtsp://rtspstream.com/parking](rtsp://rtspstream.com/parking)

<table>
  <tr>
    <td>rtsp://rtspstream.com/pattern</td>
     <td>rtsp://rtspstream.com/ball</td>
     <td>rtsp://rtspstream.com/parking</td>
  </tr>
  <tr>
    <td valign="top"><img src="https://user-images.githubusercontent.com/711/208936748-9d24526e-ed13-4a92-99a8-df6edd6a248b.png"></td>
    <td valign="top"><img src="https://user-images.githubusercontent.com/711/208938131-156808e6-3837-4a75-b0ad-f5028272dadc.gif"></td>
    <td valign="top"><img src="https://user-images.githubusercontent.com/711/208944092-94e352fa-4405-42ec-8e60-4a178733248d.gif"></td>
  </tr>
 </table>


The first two are GStreamer patterns with the current time and fly.io region overlayed on top.
The third one is a top-view parking video

While I can't guarantee reliability, you can run your own (for free, if you stay within the limits) on fly.io by following the instructions here: https://github.com/necrodome/rtsp-on-flyio

**Comment by guytechNet** [comment]
Hi necrodome,
I can not use three rtsp as follows:
rtsp://rtspstream.com/pattern
rtsp://rtspstream.com/ball
rtsp://rtspstream.com/parking

This is message error
![image](https://github.com/grigory-lobkov/rtsp-camera-view/assets/118602474/cca2a594-2daa-4704-8765-6b31e021f9ea)

**Comment by rtspstream** [comment]
@guytechNet try https://www.octostream.com for a hosted service

**Comment by necrodome** [comment]
> Hi necrodome, I can not use three rtsp as follows: rtsp://rtspstream.com/pattern rtsp://rtspstream.com/ball rtsp://rtspstream.com/parking
> 
> This is message error ![image](https://user-images.githubusercontent.com/118602474/245034378-cca2a594-2daa-4704-8765-6b31e021f9ea.png)

I suggest running your own instance where you can control the whole pipeline using https://github.com/necrodome/rtsp-on-flyio .

**Comment by USAMAWIZARD** [comment]
[Ant Media](http://antmedia.io/) An RTSP server which supports WebRTC , RTMP , HLS and Dash output.

**Comment by EEMMK** [comment]
Hey does anyone have a RTSP code from anywhere in LA I need it for a Uni project?

**Comment by pierro42** [comment]
Bump, this can help a lot of people debug their stream before thinking that the app is bugged.

**Comment by AzwadFawadHasan** [comment]
anyone knows any software that will convert a private rtsp stream into a public/http so that I can view it through the web

**Comment by USAMAWIZARD** [comment]
@AzwadFawadHasan , Yes its possible through [Ant Media](http://antmedia.io/), you can contact me at usama.tahseen@antmedia.io , would love to help you with this requirement.

**Comment by AzwadFawadHasan** [comment]
thanks i'll check it out

**Comment by tsilicani** [comment]
Most of the above URLs are no longer active.
The following it is a currently working one:
**rtsp://rtsp-test-server.viomic.com:554/stream**

**Comment by SanDyLibs** [comment]
Try these URL's :

**_rtsp://rtspstream:021ce389961591228dee593b83439e76@zephyr.rtsp.stream/movie_**
**_rtsp://rtspstream:74ded9025fe6d0ff2d1f20ec26b55591@zephyr.rtsp.stream/pattern_**

**Comment by rtspstream** [comment]
URLs from octostream.com are private URLs for the account and they have a limit of 2GB / Month of video transfer.
So, create a free account on https://www.octostream.com and use your own set of RTSP URLs.

@SanDyLibs please delete these URLs, they are not supposed to be made public and are specific to your account.

**Comment by baerrus** [comment]
> Most of the above URLs are no longer active. The following it is a currently working one: **rtsp://rtsp-test-server.viomic.com:554/stream**

Not working

**Comment by AzwadFawadHasan** [comment]
@baerrus Hi, you can check this repo. These contain some of the most stable free rtsp links out there

https://github.com/AzwadFawadHasan/Public_RTSP_Sources

**Comment by AzwadFawadHasan** [comment]
I have curated some of the working motionjpeg stream. Feel free to checkout the repo below

https://github.com/AzwadFawadHasan/Public_MotionJPEG_Sources

**Comment by baerrus** [comment]
@AzwadFawadHasan none of these links have anything to do with RTSP. They are motionJpeg streams. What am I missing? Why do you call them RTSP?

**Comment by AzwadFawadHasan** [comment]
@baerrus my bad. I'll soon fix it.

**Comment by AzwadFawadHasan** [comment]
@baerrus  I had saved some rtsp streams before. Those don't work now.  Just checked the wowza demo rtsp stream, it also stopped working for me. I guess you could create a RTSP server or use a third party media server that has rtsp output plugins.

**Comment by BraINstinct0** [comment]
@rtspstream I think it'd be best if you remove the link included in your comments (don't think it's intentional anyways)?

**Comment by rocxteady** [comment]
> @baerrus my bad. I'll soon fix it.

You still didnt delete your comments about RTSP.

**Comment by ToshibaMastr** [comment]
**rtsp://62.109.19.230:554/iloveyou**
Works only with tcp connection (ffplay -rtsp_transport tcp ...)

**Comment by ToshibaMastr** [comment]
> **rtsp://62.109.19.230:554/iloveyou** Works only with tcp connection (ffplay -rtsp_transport tcp ...)

I've updated the server, now it won't die when I'm not watching it, and you also have the opportunity to launch your streams using the command:
```sh
gst-launch-1.0 filesrc location=myaanee.mp4 ! qtdemux name=d d.video_0 ! queue ! rtspclientsink protocols=tcp name=s location=rtsp://62.109.19.230:554/myaanee d.audio_0 ! queue ! s.
```
or
```sh
ffmpeg -re -stream_loop -1 -i myaanee.mp4 -c copy -rtsp_transport tcp -f rtsp rtsp://62.109.19.230:554/myanee
```
(I don't know why, but my server doesn't want to accept udp at all)

**Comment by estate-car** [comment]
I have a pretty important thing:
Make sure that VLC is allowed on your system's firewall

**Comment by Gabka365** [comment]
Here are two probe streams for testing
https://rtsp.stream/auth/login

**Comment by ShayanFiroozi** [comment]
A beautiful video found in **2025** !

**rtsp://807e9439d5ca.entrypoint.cloud.wowza.com:1935/app-rC94792j/068b9c9a_stream2**

**Comment by tinncdev** [comment]
found an app that create either a dummy pattern or from camera, look like no ads as well.
saved me a lot of time checking all dead url above
https://apps.apple.com/vn/app/rtsp-stream/id6474928937

**Comment by iyosayi0x** [comment]
### ✅ What I Tried (Works for Testing Purposes)

* Installed **RTSP Stream** (iOS) or **RTSP Camera** (Android).
* Connected both my **phone and laptop** to the **same Wi-Fi network**.
* Created a new RTSP stream using the mobile app.
* Copied the **RTSP URL** generated by the app.
* Tested the stream successfully in **VLC Media Player**.

This setup worked perfectly for me during testing.

**Comment by rtspstream** [comment]
> found an app that create either a dummy pattern or from camera, look like no ads as well. saved me a lot of time checking all dead url above https://apps.apple.com/vn/app/rtsp-stream/id6474928937

yup, our team built that app as well, so we now have both an iOS app ([OctoStream RTSP Streamer](https://apps.apple.com/au/app/octostream-rtsp-streamer/id6474928937)) as well as hosted RTSP streams on https://www.octostream.com. Both are free. The website has bandwidth caps to make sure the servers are not overwhelmed. Use whatever works best. And please let us know if you guys are looking for more features. We would be happy to add more support.

**Comment by volkov-maxim** [comment]
Another way: create RTSP stream for test from Youtube video/live with [go2rtc](https://github.com/AlexxIT/go2rtc).
[In my case, I did it for Frigate NVR](https://share.google/aimode/LHjyP79knlvdbetJO).
